e2etests/tests/unnumbered.go

```diff
@@ -0,0 +1,285 @@
+			ginkgo.Skip("Skipping this test because RUN_FRR_CONTAINER_ON_HOST_NETWORK is set to true")
+		}
+		var err error
+		updater, err = config.NewUpdater()
+		Expect(err).NotTo(HaveOccurred())
+
+		err = updater.Clean()
+		Expect(err).NotTo(HaveOccurred())
+
+		cs = k8sclient.New()
+		nodes, err := k8s.Nodes(cs)
+		Expect(err).NotTo(HaveOccurred())
+		nodeWithP2PConnection = nodes[0]
+
+	})
+
+	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentSpecReport().Failed() {
+			reporter := dump.NewK8sReporter(k8s.FRRK8sNamespace)
+			testName := ginkgo.CurrentSpecReport().FullText()
+			dump.K8sInfo(testName, reporter)
+			dump.BGPInfo(testName, []*frrcontainer.FRR{externalP2PContainer}, cs)
+		}
+	})
+
+	ginkgo.DescribeTable("BGP session is established and routes are verified", func(rc frrconfig.RouterConfigUnnumbered) {
+		iface := rc.Interface
+		remoteASN := rc.ASNLocal
+
+		ginkgo.By(fmt.Sprintf("creating p2p %s:%s -- %s:external-p2p-container", nodeWithP2PConnection.Name, iface, iface))
+		externalP2PContainer, err := frrcontainer.CreateP2PPeerFor(nodeWithP2PConnection.Name, iface, FRRImage)
+		Expect(err).NotTo(HaveOccurred())
+		ginkgo.DeferCleanup(func() {
+			err := frrcontainer.Delete([]*frrcontainer.FRR{externalP2PContainer})
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		ginkgo.By(fmt.Sprintf("updating frrconfig to %s", externalP2PContainer.Name))
+		c, err := rc.Config()
+		Expect(err).NotTo(HaveOccurred())
+		err = externalP2PContainer.UpdateBGPConfigFile(c)
+		Expect(err).NotTo(HaveOccurred())
+
+		ginkgo.By(fmt.Sprintf("peering node %s with its p2p container %s", nodeWithP2PConnection.Name, externalP2PContainer.Name))
+		cr := frrk8sv1beta1.FRRConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "for-" + externalP2PContainer.Name,
+				Namespace: k8s.FRRK8sNamespace,
+			},
+			Spec: frrk8sv1beta1.FRRConfigurationSpec{
+				BGP: frrk8sv1beta1.BGPConfig{
+					BFDProfiles: []frrk8sv1beta1.BFDProfile{{Name: "simple"}},
+					Routers: []frrk8sv1beta1.Router{
+						{
+							ASN: infra.FRRK8sASN,
+							Neighbors: []frrk8sv1beta1.Neighbor{
+								{
+									ASN:        remoteASN,
+									Interface:  iface,
+									BFDProfile: "simple",
+									ToReceive: frrk8sv1beta1.Receive{
+										Allowed: frrk8sv1beta1.AllowedInPrefixes{
+											Mode: frrk8sv1beta1.AllowAll,
+										},
+									},
+									ToAdvertise: frrk8sv1beta1.Advertise{
+										Allowed: frrk8sv1beta1.AllowedOutPrefixes{
+											Mode: frrk8sv1beta1.AllowAll,
+										},
+									},
+								},
+							},
+							Prefixes: prefixSend,
+						},
+					},
+				},
+				NodeSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"kubernetes.io/hostname": nodeWithP2PConnection.GetLabels()["kubernetes.io/hostname"],
+					},
+				},
+			},
+		}
+		err = updater.Update([]corev1.Secret{}, cr)
+		Expect(err).NotTo(HaveOccurred())
+		ginkgo.DeferCleanup(func() {
+			err := updater.DeleteFRRConfiguration(cr)
```

Comment:
why do we need this and the clean we are doing in beforeEach is not enough?

Reply:
the clean in beforeEach is generic clean all so is enough and currently required because most tests do not cleanup their CRs. Nevertheless from gikngko perspective it is recommended each test to cleanup the resources, for example if we want to move towards parallel test that is a small contribution.

Given you comment I will remove it also.

Reply:
What's bugging me is that all the tests clean the configuration to start with a clean slate, then I land on this and have to spend extra energy to understand why it's different, just to find out that it's unnecessary.
So yes, let's keep the structure consistent with the other tests.

Reply:
removed  for consistency 

(but for the record is bugging me the opposite, why tests are not take care of the cleanup)